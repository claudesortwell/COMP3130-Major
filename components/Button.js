import React from "react";
import { Text, TouchableHighlight, StyleSheet, View } from "react-native";
import { LinearGradient } from "expo-linear-gradient";
import Shadows from "../styles/shadow";
import Colors from "../styles/colors";

export const Button = ({ type, onPress, text, icon, iconRight }) => {
  if (type === "primary") {
    return (
      <TouchableHighlight
        style={{ borderRadius: 5 }}
        onPress={() => {
          if (onPress) onPress();
        }}
      >
        <View style={{ ...Shadows.main }}>
          <LinearGradient
            colors={[Colors.teal, Colors.blue]}
            style={{ ...styles.buttonPrimary, ...Shadows.main }}
            start={{ x: 0, y: 0 }}
            end={{ x: 1, y: 1 }}
          >
            <Text style={styles.buttonTextPrimary}>
              {iconRight}
              {"   "}
              {text}
              {"   "}
              {icon}
            </Text>
          </LinearGradient>
        </View>
      </TouchableHighlight>
    );
  } else {
    return (
      <TouchableHighlight
        style={{ borderRadius: 5 }}
        onPress={() => {
          if (onPress) onPress();
        }}
      >
        <View style={{ ...styles.buttonSecondary, ...Shadows.main }}>
          <Text style={styles.buttonTextSecondary}>
            {iconRight}
            {"   "}
            {text}
            {"   "}
            {icon}
          </Text>
        </View>
      </TouchableHighlight>
    );
  }
};

const styles = StyleSheet.create({
  buttonPrimary: {
    backgroundColor: Colors.blue,
    padding: 16,
    borderRadius: 5
  },
  buttonTextPrimary: {
    fontFamily: "Inter_600SemiBold",
    textAlign: "center",
    fontSize: 14,
    color: Colors.white
  },
  buttonSecondary: {
    backgroundColor: Colors.white,
    padding: 16,
    borderRadius: 5
  },
  buttonTextSecondary: {
    fontFamily: "Inter_600SemiBold",
    textAlign: "center",
    fontSize: 14,
    color: Colors.main
  }
});
