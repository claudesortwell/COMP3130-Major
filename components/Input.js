import React from "react";
import { Text, StyleSheet, View, TextInput } from "react-native";
import TextStyle from "../styles/text";
import Colors from "../styles/colors";
import { AntDesign } from "@expo/vector-icons";

export const Input = ({ label, value, placeholder, setValue, icon, whiteLabel, error, hint, ...otherProps }) => {
  return (
    <View style={{ width: "100%" }}>
      <Text style={{ ...TextStyle.H3, color: whiteLabel ? Colors.white : Colors.main, marginBottom: 8 }}>
        {icon && (
          <>
            {icon}
            {"   "}
          </>
        )}

        {label}
      </Text>
      <TextInput
        style={{ ...styles.main, borderColor: error ? Colors.red : Colors.grey }}
        onChangeText={(text) => setValue(text)}
        placeholder={placeholder}
        value={value}
        {...otherProps}
      />
      {error && (
        <View
          style={{
            marginTop: 3,
            borderRadius: 5,
            backgroundColor: whiteLabel ? "#ffffff" : "",
            padding: 3
          }}
        >
          <Text
            style={{
              ...TextStyle.H4,
              color: Colors.red
            }}
          >
            <AntDesign name="exclamationcircle" size={14} />
            {"  "}
            {error}
          </Text>
        </View>
      )}
      {hint && !error && <Text style={{ ...TextStyle.H4, color: Colors.main, marginTop: 8 }}>{hint}</Text>}
    </View>
  );
};

const styles = StyleSheet.create({
  main: {
    height: 50,
    padding: 10,
    width: "100%",
    borderRadius: 5,
    backgroundColor: Colors.white,
    borderWidth: 1
  },
  buttonTextPrimary: {
    fontFamily: "Inter_600SemiBold",
    textAlign: "center",
    fontSize: 14,
    color: Colors.white
  },
  buttonSecondary: {
    backgroundColor: Colors.white,
    padding: 16,
    borderRadius: 5
  },
  buttonTextSecondary: {
    fontFamily: "Inter_600SemiBold",
    textAlign: "center",
    fontSize: 14,
    color: Colors.main
  }
});
