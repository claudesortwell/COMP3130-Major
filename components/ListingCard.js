import React, { useState } from "react";
import { View, Image, Text, TouchableHighlight } from "react-native";
import Shadows from "../styles/shadow";
import TextStyle from "../styles/text";
import { MaterialCommunityIcons } from "@expo/vector-icons";
import { AntDesign } from "@expo/vector-icons";
import { Entypo } from "@expo/vector-icons";
import { MaterialIcons } from "@expo/vector-icons";
import { LinearGradient } from "expo-linear-gradient";
import Colors from "../styles/colors";

const ListingCard = ({ data, navigation }) => {
  return (
    <View
      style={{
        backgroundColor: "#fff",
        width: "100%",
        height: 175,
        borderRadius: 5,
        ...Shadows.main,
        margin: 10,
        flex: 1,
        flexDirection: "row"
      }}
    >
      <Image
        style={{ width: "50%", height: 175, borderTopLeftRadius: 5, borderBottomLeftRadius: 5 }}
        source={data.image.toString().startsWith("file:/") ? { uri: data.image } : data.image}
      />
      <View style={{ padding: 10, flex: 1 }}>
        <Text numberOfLines={1} style={{ ...TextStyle.H3 }}>
          {data.name}
        </Text>

        <View style={{ flexDirection: "row", marginTop: 8 }}>
          <MaterialCommunityIcons name="map-marker" size={18} />
          <Text numberOfLines={2} style={{ ...TextStyle.H6, paddingLeft: 7, marginBottom: 7, width: "80%" }}>
            {data.address}
          </Text>
        </View>

        <View style={{ flexDirection: "row", marginTop: data.locality ? 2 : 16 }}>
          <AntDesign name="tags" size={18} />
          <Text
            numberOfLines={1}
            style={{ ...TextStyle.H6, paddingLeft: 7, paddingTop: 3, marginBottom: 7, width: "85%" }}
          >
            {data.category}
          </Text>
        </View>

        {data.locality && (
          <View style={{ flexDirection: "row", marginTop: 10 }}>
            <Entypo name="globe" size={14} style={{ paddingLeft: 2 }} />
            <Text numberOfLines={1} style={{ ...TextStyle.H6, paddingLeft: 9, marginTop: 1.5, width: "85%" }}>
              {data.locality}
            </Text>
          </View>
        )}

        <View style={{ flexDirection: "row", marginTop: data.locality ? 10 : 16 }}>
          <MaterialIcons name="star" size={18} color="black" />
          <Text
            numberOfLines={1}
            style={{ ...TextStyle.H6, paddingLeft: 7, paddingTop: 3, marginBottom: 7, width: "85%" }}
          >
            {data.stars}/5
          </Text>
        </View>

        <View style={{ flexDirection: "row" }}>
          <TouchableHighlight
            style={{ borderRadius: 5, flexDirection: "row", marginTop: 3 }}
            onPress={() => {
              if (navigation) navigation.navigate("Listing", { id: data.id });
            }}
          >
            <LinearGradient
              colors={[Colors.teal, Colors.blue]}
              style={{ ...Shadows.main, borderRadius: 5, width: "100%" }}
              start={{ x: 0, y: 0 }}
              end={{ x: 1, y: 1 }}
            >
              <Text style={{ ...TextStyle.H5, color: Colors.white, padding: 5, textAlign: "center" }}>View More</Text>
            </LinearGradient>
          </TouchableHighlight>
        </View>
      </View>
    </View>
  );
};

export default ListingCard;
