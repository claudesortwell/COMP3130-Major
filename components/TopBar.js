import React, { useEffect } from "react";
import { Text, View, Image, TouchableWithoutFeedback, TouchableOpacity, StyleSheet, Modal } from "react-native";
import TextStyle from "../styles/text";
import { Entypo } from "@expo/vector-icons";
import { MaterialIcons } from "@expo/vector-icons";
import Shadows from "../styles/shadow";
import DataManager from "../data/DataManager";

export const TopBar = ({ navigation, disableShadow }) => {
  const [menuOpen, setMenuOpen] = React.useState(false);
  const [dataManager, setDataManager] = React.useState(null);

  useEffect(() => {
    DataManager.getInstance().then((instance) => setDataManager(instance));
  }, []);

  return (
    <View
      style={
        disableShadow
          ? {
              backgroundColor: "#fff",
              height: 100,
              paddingTop: 50,
              paddingLeft: 30,
              paddingRight: 20,
              flexDirection: "row",
              flexWrap: "wrap",
              zIndex: 20
            }
          : {
              backgroundColor: "#fff",
              height: 100,
              paddingTop: 50,
              paddingLeft: 30,
              paddingRight: 20,
              borderBottomLeftRadius: 20,
              borderBottomRightRadius: 20,
              flexDirection: "row",
              flexWrap: "wrap",
              zIndex: 20,
              ...Shadows.main
            }
      }
    >
      <Image source={require("../assets/logo.png")} style={{ width: 30, height: 30 }} />
      <Image
        resizeMode={"contain"}
        source={require("../assets/SydneyCity.png")}
        style={{ width: 80, height: 30, marginLeft: 10, marginTop: 2 }}
      />
      <TouchableWithoutFeedback onPress={() => setMenuOpen(!menuOpen)}>
        <View
          style={{
            position: "absolute",
            right: 0,
            paddingTop: 55,
            paddingLeft: 20,
            paddingRight: 30,
            zIndex: 100
          }}
        >
          <Entypo name="dots-three-vertical" size={20} color="black" />
        </View>
      </TouchableWithoutFeedback>
      <Modal transparent visible={menuOpen} animationType="fade" onRequestClose={() => {}}>
        <TouchableOpacity
          activeOpacity={1}
          onPress={() => {
            dataManager.logout().then(() => {
              navigation.navigate("Welcome");
              setMenuOpen(false);
            });
          }}
          style={{
            backgroundColor: "transparent",
            flex: 1
          }}
        >
          <View
            style={{
              width: 120,
              borderRadius: 10,
              right: 0,
              position: "absolute",
              marginTop: 80,
              padding: 10,
              marginRight: 30,
              backgroundColor: "#fff",
              ...Shadows.main
            }}
          >
            <Text style={{ ...TextStyle.H3, textAlign: "center" }}>
              <MaterialIcons size={14} name="logout" />
              {"  "}
              Logout
            </Text>
          </View>
        </TouchableOpacity>
      </Modal>
    </View>
  );
};
