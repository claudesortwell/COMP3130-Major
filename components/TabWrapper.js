import React from "react";
import { Text, View, StyleSheet } from "react-native";
import { TopBar } from "../components/TopBar";
import { ScrollView } from "react-native-gesture-handler";

const styles = StyleSheet.create({
  mainContainer: {
    flex: 1,
    zIndex: 10
  }
});

export const TabWrapper = ({ navigation, children }) => {
  const { mainContainer } = styles;

  return (
    <View style={mainContainer}>
      <TopBar navigation={navigation} />
      <ScrollView style={{ zIndex: 5 }}>{children}</ScrollView>
    </View>
  );
};
