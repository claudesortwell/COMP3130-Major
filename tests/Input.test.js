import React from "react";
import renderer from "react-test-renderer";
import { Input } from "../components/Input";
jest.mock("@expo/vector-icons");

describe("<Input />", () => {
  it("Renders Basic input", () => {
    const renderInput = renderer.create(
      <Input
        keyboardType={"email-address"}
        label={"Email"}
        autoCapitalize="none"
        placeholder={"Enter your account email address"}
        whiteLabel={true}
      />
    );

    expect(renderInput).toMatchSnapshot();
  });

  it("Input without white label enabled", () => {
    const renderInput = renderer.create(
      <Input
        keyboardType={"email-address"}
        label={"Email"}
        autoCapitalize="none"
        placeholder={"Enter your account email address"}
        whiteLabel={false}
      />
    );

    expect(renderInput).toMatchSnapshot();
  });

  it("Input with errors (touched and untouched)", () => {
    const renderInput = renderer.create(
      <Input
        keyboardType={"email-address"}
        label={"Email"}
        autoCapitalize="none"
        placeholder={"Enter your account email address"}
        whiteLabel={false}
        error={"Email is invalid"}
      />
    );

    const renderInputTouched = renderer.create(
      <Input
        keyboardType={"email-address"}
        label={"Email"}
        autoCapitalize="none"
        placeholder={"Enter your account email address"}
        whiteLabel={false}
        error={"Email is invalid"}
        touched={true}
      />
    );

    expect(renderInput).toMatchSnapshot();
    expect(renderInputTouched).toMatchSnapshot();
  });

  it("Input with errors (touched and untouched)", () => {
    const renderInputHintAndError = renderer.create(
      <Input
        keyboardType={"email-address"}
        label={"Email"}
        autoCapitalize="none"
        placeholder={"Enter your account email address"}
        whiteLabel={false}
        error={"Email is invalid"}
        hint={"Email is invalid"}
        touched={true}
      />
    );

    const renderInputHint = renderer.create(
      <Input
        keyboardType={"email-address"}
        label={"Email"}
        autoCapitalize="none"
        placeholder={"Enter your account email address"}
        whiteLabel={false}
        hint={"Email is invalid"}
      />
    );

    // Error should show instead of hint  because error is enabled
    expect(renderInputHintAndError.toJSON().children[2].type).toEqual("View");

    // Hint should show in a text component
    expect(renderInputHint.toJSON().children[2].type).toEqual("Text");
  });
});
