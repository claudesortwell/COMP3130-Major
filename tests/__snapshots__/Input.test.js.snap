// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> Input with errors (touched and untouched) 1`] = `
<View
  style={
    Object {
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#252F3F",
        "fontFamily": "Inter_700Bold",
        "fontSize": 14,
        "marginBottom": 8,
      }
    }
  >
    Email
  </Text>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    keyboardType="email-address"
    onChangeText={[Function]}
    placeholder="Enter your account email address"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#fff",
        "borderColor": "#D2D6DC",
        "borderRadius": 5,
        "borderWidth": 1,
        "height": 50,
        "padding": 10,
        "width": "100%",
      }
    }
    underlineColorAndroid="transparent"
  />
</View>
`;

exports[`<Input /> Input with errors (touched and untouched) 2`] = `
<View
  style={
    Object {
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#252F3F",
        "fontFamily": "Inter_700Bold",
        "fontSize": 14,
        "marginBottom": 8,
      }
    }
  >
    Email
  </Text>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    keyboardType="email-address"
    onChangeText={[Function]}
    placeholder="Enter your account email address"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#fff",
        "borderColor": "#F05252",
        "borderRadius": 5,
        "borderWidth": 1,
        "height": 50,
        "padding": 10,
        "width": "100%",
      }
    }
    underlineColorAndroid="transparent"
  />
  <View
    style={
      Object {
        "backgroundColor": "",
        "borderRadius": 5,
        "marginTop": 3,
        "padding": 3,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#F05252",
          "fontFamily": "Inter_600SemiBold",
          "fontSize": 14,
        }
      }
    >
        
      Email is invalid
    </Text>
  </View>
</View>
`;

exports[`<Input /> Input without white label enabled 1`] = `
<View
  style={
    Object {
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#252F3F",
        "fontFamily": "Inter_700Bold",
        "fontSize": 14,
        "marginBottom": 8,
      }
    }
  >
    Email
  </Text>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    keyboardType="email-address"
    onChangeText={[Function]}
    placeholder="Enter your account email address"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#fff",
        "borderColor": "#D2D6DC",
        "borderRadius": 5,
        "borderWidth": 1,
        "height": 50,
        "padding": 10,
        "width": "100%",
      }
    }
    underlineColorAndroid="transparent"
  />
</View>
`;

exports[`<Input /> Renders Basic input 1`] = `
<View
  style={
    Object {
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontFamily": "Inter_700Bold",
        "fontSize": 14,
        "marginBottom": 8,
      }
    }
  >
    Email
  </Text>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    keyboardType="email-address"
    onChangeText={[Function]}
    placeholder="Enter your account email address"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#fff",
        "borderColor": "#D2D6DC",
        "borderRadius": 5,
        "borderWidth": 1,
        "height": 50,
        "padding": 10,
        "width": "100%",
      }
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
